expansion/src/main/java/no/unit/nva/expansion/model/ExpandedResource.java

```diff
@@ -74,12 +73,44 @@ public static ExpandedResource fromPublication(RawContentRetriever uriRetriever,
         return attempt(() -> objectMapper.treeToValue(sortedJson, ExpandedResource.class)).orElseThrow();
     }
 
-    private static JsonNode addFields(String json, Publication publication) {
-        var sortedJson = strToJsonWithSortedContributors(json);
-        injectHasFileEnum(publication, (ObjectNode) sortedJson);
-        expandLicenses(sortedJson);
-        return sortedJson;
-    }
+  private static JsonNode addFields(String json, Publication publication) {
+    var sortedJson = strToJsonWithSortedContributors(json);
+    injectHasFileEnum(publication, (ObjectNode) sortedJson);
+    expandLicenses(sortedJson);
+    injectAnthologyRelation(publication, (ObjectNode) sortedJson);
+    return sortedJson;
+  }
+
+  private static void injectAnthologyRelation(Publication publication, ObjectNode sortedJson) {
+    Optional.ofNullable(publication.getEntityDescription())
+        .flatMap(desc -> Optional.ofNullable(desc.getReference()))
+        .ifPresent(
+            reference -> {
+              var instanceType = reference.getPublicationInstance();
+              var publicationContext = reference.getPublicationContext();
+              var isPartOfAnthology =
+                  publicationContext instanceof Anthology
+                      && (instanceType instanceof ChapterArticle
+                          || instanceType instanceof ChapterConferenceAbstract
+                          || instanceType instanceof ChapterInReport);
+
+              if (instanceType instanceof BookAnthology) {
+                addJoinField(sortedJson, "hasParts", null);
+              } else if (isPartOfAnthology) {
+                var parentId = ((Anthology) publicationContext).getId();
+                var parentIdentifier = SortableIdentifier.fromUri(parentId).toString();
+                addJoinField(sortedJson, "partOf", parentIdentifier);
+              }
+            });
+  }
```

Comment:
Maybe extract as method, for example:
.ifPresent(reference -> injectJoinFieldWhenAnthology(sortedJson, reference))